packages/dataviews/src/field-types/number.tsx

```diff
@@ -24,6 +29,49 @@ import {
 import RenderFromElements from './utils/render-from-elements';
 import sort from './utils/sort-number';
 
+function getFormat< Item >( field: Field< Item > ): Required< FormatNumber > {
+	const fieldFormat = field.format as FormatNumber | undefined;
+	return {
+		separatorThousand:
+			fieldFormat?.separatorThousand !== undefined &&
+			typeof fieldFormat.separatorThousand === 'string'
+				? fieldFormat.separatorThousand
+				: ',',
+		separatorDecimal:
+			fieldFormat?.separatorDecimal !== undefined &&
+			typeof fieldFormat.separatorDecimal === 'string'
+				? fieldFormat.separatorDecimal
+				: '.',
+		decimals:
+			fieldFormat?.decimals !== undefined &&
+			typeof fieldFormat.decimals === 'number' &&
+			fieldFormat.decimals >= 0 &&
+			fieldFormat.decimals <= 100 &&
+			Number.isInteger( fieldFormat.decimals )
+				? fieldFormat.decimals
+				: 2,
+	};
+}
+
+export function formatNumber(
+	value: number,
+	format: Required< FormatNumber >
+): string {
+	if ( ! Number.isFinite( value ) ) {
+		return String( value );
+	}
+	const { separatorThousand, separatorDecimal, decimals } = format;
+	const fixedValue = value.toFixed( decimals );
+	const [ integerPart, decimalPart ] = fixedValue.split( '.' );
+	const formattedInteger = integerPart.replace(
```

Comment:
In format integer we have:
if ( ! separatorThousand ) {
    return String( integerValue );
}

I guess we could also add it to formatNumber for consistency.

Reply:
I guess you meant avoiding the regex replace if `! separatorThousand`, right? I did that [here](https://github.com/WordPress/gutenberg/pull/73644/commits/180846297c98a9ae4a6fd70e9702eba1b073362d).